packages/hop-node/src/watchers/classes/Bridge.ts

```diff
@@ -752,7 +755,24 @@ export default class Bridge extends ContractBase {
       gasPrice = gasPrice.div(2)
       gasLimit = gasLimit.div(2)
     }
-    const gasCost = gasLimit.mul(gasPrice)
+    let gasCost = gasLimit.mul(gasPrice)
+
+    if (this.chainSlug === Chain.Optimism && data && to) {
+      try {
+        const ovmGasPriceOracle = getContractFactory('OVM_GasPriceOracle')
+          .attach(predeploys.OVM_GasPriceOracle).connect(getRpcProvider(this.chainSlug))
+        const serializedTx = serializeTransaction({
+          value: parseEther('0'),
+          gasPrice,
+          gasLimit,
+          to,
+          data
+        })
+        gasCost = await ovmGasPriceOracle.getL1Fee(serializedTx)
```

Comment:
I think this value should be added to the existing `gasCost`. IIUC, this value provides the L1 cost, in Wei and needs to be added to our existing calc, `gasCost = gasLimit.mul(gasPrice)`, which is the L2 cost.

Reply:
Fixed with https://github.com/hop-protocol/hop/pull/287/commits/5f60e11bb197451ff9f6d15ea3edc644dab3f5b0